web/src/main/java/com/microsoft/applicationinsights/web/internal/WebAppInitializer.java

```diff
@@ -0,0 +1,165 @@
+        }
+    }
+
+    @Override
+    public void contextDestroyed(ServletContextEvent sce) {
+    }
+
+    private void addFilter(ServletContext context) {
+        FilterRegistration filterData = getAIFilter(context);
+        if (filterData == null) {
+            // Adding the filter since we didn't find that in configuration.
+
+            WebRequestTrackingFilter filter = new WebRequestTrackingFilter();
+            FilterRegistration.Dynamic filterRegistration = context.addFilter(FILTER_NAME, filter);
+
+            if (filterRegistration == null) {
+                // This should not happen
+                InternalLogger.INSTANCE.logAlways(InternalLogger.LoggingLevel.ERROR, "Failed to add filter '%s'", FILTER_NAME);
+
+                return;
+            }
+
+            filterRegistration.addMappingForUrlPatterns(null, false, "/*");
+        } else {
+            filterData.addMappingForUrlPatterns(null, false, "/*");
+        }
+    }
+
+    private String registerWebApp(String name) {
+        String key = null;
+
+        if (!Strings.isNullOrEmpty(name)) {
+            InternalLogger.INSTANCE.logAlways(InternalLogger.LoggingLevel.INFO, "Registering WebApp with name '%s'", name);
+            key = AgentConnector.INSTANCE.register(this.getClass().getClassLoader(), name);
+            if (Strings.isNullOrEmpty(key)) {
+                InternalLogger.INSTANCE.logAlways(InternalLogger.LoggingLevel.ERROR, "Key for '%s' key is null'. No way to have RDD telemetries for this WebApp", name);
+            } else {
+                InternalLogger.INSTANCE.logAlways(InternalLogger.LoggingLevel.INFO, "Registered WebApp '%s' key='%s'", name, key);
+            }
+        } else {
+            InternalLogger.INSTANCE.logAlways(InternalLogger.LoggingLevel.ERROR, "WebApp name is not found, unable to register WebApp");
+        }
+
+        return key;
+    }
+
+    private FilterRegistration getAIFilter(ServletContext context){
+        String filterClassName = WebRequestTrackingFilter.class.getName();
+
+        Collection<? extends FilterRegistration> filterRegistrations = context.getFilterRegistrations().values();
+        for ( FilterRegistration filterRegistration : filterRegistrations) {
+            if (filterClassName.equals(filterRegistration.getClassName())) {
+                InternalLogger.INSTANCE.logAlways(InternalLogger.LoggingLevel.INFO, "AI-Filter is already defined");
+
+                return filterRegistration;
+            }
+        }
+
+        InternalLogger.INSTANCE.logAlways(InternalLogger.LoggingLevel.INFO, "Did not find AI-Filter");
+        return null;
+    }
+
+    private String getName(ServletContext context) {
+        String name = null;
+        try {
+            String contextPath = context.getContextPath();
+            if (Strings.isNullOrEmpty(contextPath)) {
```

Comment:
I know. I thought that we should start with this, later on we can add support for explicit naming through configuration,